docs/tutorials/quesion-answering-bot.md

```diff
@@ -0,0 +1,223 @@
+
+Now that you have a general idea of what we will be doing, the following section will show you how to define our `Flow`s in Jina. Then we will take a look at how to implement the necessary `Executor`s for our search-based question-answering system.  
+
+## Indexing the text document 
+Let's imagine we extracted a bunch of sentences from Jina's documentation and stored them in a `DocumentArray`, as shown below. 
+
+```python
+example_sentences = [
+    "Document is the basic data type that Jina operates with",
+    "Executor processes a DocumentArray in-place", 
+    ...,
+    "Jina uses the concept of a flow to tie different executors together"
+]
+
+docs = DocumentArray([Document(content=sentence) for sentence in example_sentences])
+```
+
+As described in the last section, we first need to predict potential questions for each of the elements in the `DocumentArray`. Then we have to use another model to create vector encodings from the predicted questions. Finally, we store them as the index. 
+
+At this point we have enough information to start defining our `Flows`.
+
+*Without further ado, let's build!*
+
+``` python
+indexing_flow = Flow(
+# Generate potential questions using doc2query
+).add(name="question_transformer", 
+      uses=QuestionGenerator, 
+      uses_with={"random_seed": 12345}
+# Create vector representations for generated questions 
+).add(name="text_encoder", 
+      uses=TextEncoder, 
+      uses_with={"parameters": {"traversal_paths": ["c"]}}
+# Store embeddings for all generated questions as index
+).add(name="my_indexer", 
+      uses=MyIndexer
+)
+
+with indexing_flow: 
+    # Run the indexing on all extracted sentences
+    indexing_flow.post(on="/index", inputs=docs, on_done=print)
+```
+
+## Searching of the user's query against the index
+
+After having defined the `Flow` for indexing our document, we are now ready to work on answering user queries. Incoming queries also need to be encoded. For that, we use the same encoder that we used for encoding our generated questions. Then we need `SimpleIndexer` to perform similarity search, in order to retrieve generated questions and eventually answers the query. 
+
+The flow for searching is much simpler than the one for indexing and looks like this: 
+
+``` python
+query_flow = Flow(
+    # Create vector representations from query
+    ).add(name="query_transformer", uses=TextEncoder
+    # Search the index for matching generated questions
+    ).add(name="query_indexer", uses=MyIndexer)
+
+with query_flow: 
+    indexing_flow.post(on="/query", inputs=user_queries, on_done=print)
+```
+
+Now that we have seen the overall structure of the approach and have defined our `Flows`, we can code up the `Executor`s.
+
+## Building the Executor to Generate Potential Questions 
+
+The first `Executor`, that we implement, is the `QuestionGenerator`. It is basically a wrapper around the model that predicts potential questions, which a given piece of text can answer.
```

Comment:
```suggestion
The first `Executor` that we implement is the `QuestionGenerator`. It is a wrapper around the model that predicts potential questions, which a given piece of text can answer.
```

Reply:
You are right. `Basically` should be removed. The commas however, need to be there.